xbmc/addons/kodi-dev-kit/include/kodi/tools/EndTime.h

```diff
@@ -0,0 +1,198 @@
+  ///
+  inline explicit CEndTime(unsigned int millisecondsIntoTheFuture)
+    : m_startTime(std::chrono::system_clock::now().time_since_epoch()),
+      m_totalWaitTime(std::chrono::milliseconds(millisecondsIntoTheFuture))
+  {
+  }
+
+  //============================================================================
+  /// @ingroup cpp_kodi_tools_CEndTime
+  /// @brief Set the time in the future we cosider this timer as expired
+  ///
+  /// @param[in] millisecondsIntoTheFuture the time in the future we cosider this timer as expired
+  ///
+  inline void Set(unsigned int millisecondsIntoTheFuture)
+  {
+    using namespace std::chrono;
+
+    m_startTime = system_clock::now().time_since_epoch();
+    m_totalWaitTime = milliseconds(millisecondsIntoTheFuture);
+  }
+
+  //============================================================================
+  /// @ingroup cpp_kodi_tools_CEndTime
+  /// @brief Check if the expiry time has been reached
+  ///
+  /// @return True if the expiry amount of time has past, false otherwise
+  ///
+  inline bool IsTimePast() const
+  {
+    using namespace std::chrono;
+
+    return m_totalWaitTime.count() == std::numeric_limits<unsigned int>::max()
+               ? false
+               : (m_totalWaitTime.count() == 0
+                      ? true
+                      : (system_clock::now().time_since_epoch() - m_startTime) >= m_totalWaitTime);
+  }
+
+  //============================================================================
+  /// @ingroup cpp_kodi_tools_CEndTime
+  /// @brief The amount of time left till this timer expires
+  ///
+  /// @return 0 if the expiry amount of time has past, the numbe rof milliseconds remaining otherwise
+  ///
+  inline unsigned int MillisLeft() const
+  {
+    using namespace std::chrono;
+
+    if (m_totalWaitTime.count() == std::numeric_limits<unsigned int>::max())
+      return std::numeric_limits<unsigned int>::max();
+    if (m_totalWaitTime.count() == 0)
+      return 0;
+    unsigned int timeWaitedAlready =
+        duration_cast<milliseconds>(std::chrono::system_clock::now().time_since_epoch() -
+                                    m_startTime)
+            .count();
+    return (timeWaitedAlready >= m_totalWaitTime.count())
+               ? 0
+               : (m_totalWaitTime.count() - timeWaitedAlready);
+  }
+
+  //============================================================================
+  /// @ingroup cpp_kodi_tools_CEndTime
+  /// @brief Consider this timer expired
+  ///
+  inline void SetExpired()
+  {
+    using namespace std::chrono;
+    m_totalWaitTime = milliseconds(0);
+  }
+  //============================================================================
+  /// @ingroup cpp_kodi_tools_CEndTime
+  /// @brief Set this timer as never expiring
+  ///
+  inline void SetInfinite()
+  {
+    using namespace std::chrono;
+    m_totalWaitTime = milliseconds(std::numeric_limits<unsigned int>::max());
+  }
```

Comment:
Add empty lines behind. Also has I used after a function always a //-------... to define end (is mostly by mind and not mandatory).